File: sd/trabalhoFinal/final/src/client/cabecalho.py
from json import *

class Cabecalho:

    def __init__(self, messageType, objectreference, methodId, arguments, requestId):
        self.messageType = messageType
        self.objectreference = objectreference
        self.methodId = methodId
        self.arguments = arguments
        self.requestId = requestId

    def to_json(self):
        headerData = {
            "messageType": self.messageType,
            "objectreference": self.objectreference,
            "methodId": self.methodId,
            "arguments": self.arguments,
            "requestId": self.requestId
        }
        return dumps(headerData)

    @classmethod
    def from_json(cls, json_str):
        headerData = loads(json_str)
        return cls(
            headerData["messageType"],
            headerData["objectreference"],
            headerData["methodId"],
            headerData["arguments"],
            headerData.get("requestId", 0)
        )

File: sd/trabalhoFinal/final/src/server/Cabecalho.java
package org.example;
import org.json.JSONObject;

public class Cabecalho {
    private int messageType;
    private String objectReference;
    private int methodId;
    private String arguments;

    public Cabecalho(int messageType, String objectReference, int methodId, String arguments) {
        this.messageType = messageType;
        this.objectReference = objectReference;
        this.methodId = methodId;
        this.arguments = arguments;
    }

    public String toJson() {
        JSONObject headerData = new JSONObject();
        headerData.put("messageType", messageType);
        headerData.put("objectreference", objectReference);
        headerData.put("methodId", methodId);
        headerData.put("arguments", arguments);
        return headerData.toString();
    }

    public static Cabecalho fromJson(String jsonStr) {
        JSONObject headerData = new JSONObject(jsonStr);
        return new Cabecalho(
                headerData.getInt("messageType"),
                headerData.getString("objectreference"),
                headerData.getInt("methodId"),
                headerData.get("arguments").toString()
        );
    }

    public String getObjectReference() {
        return objectReference;
    }

    public int getMethodId() {
        return methodId;
    }

    public String getArguments() {
        return arguments;
    }
}